widget/src/lazy/component.rs

```diff
@@ -254,11 +254,12 @@ where
 
     fn layout(
         &self,
+        tree: &mut Tree,
         renderer: &Renderer,
         limits: &layout::Limits,
     ) -> layout::Node {
         self.with_element(|element| {
-            element.as_widget().layout(renderer, limits)
+            element.as_widget().layout(tree, renderer, limits)
```

Comment:
We need to use the same logic for accessing `tree` as in `on_event`, this currently panics when using `Component`.

Reply:
Good catch! Thanks!

Reply:
I have a feeling that the lazy widgets may be able to be simplified a bunch now that we have access to a mutable `Tree` in `layout` (`Overlay::layout` also takes a mutable self now).